src/ray/raylet/node_manager.cc

```diff
@@ -1933,23 +1933,28 @@ void NodeManager::HandleDrainRaylet(rpc::DrainRayletRequest request,
                                     rpc::SendReplyCallback send_reply_callback) {
   RAY_LOG(INFO) << "Drain raylet RPC has received. Drain reason: "
                 << request.reason_message();
-
-  if (request.reason() ==
-      rpc::autoscaler::DrainNodeReason::DRAIN_NODE_REASON_IDLE_TERMINATION) {
-    const bool is_idle =
-        cluster_resource_scheduler_->GetLocalResourceManager().IsLocalNodeIdle();
-    if (is_idle) {
-      cluster_resource_scheduler_->GetLocalResourceManager().SetLocalNodeDraining();
+  switch (request.reason()) {
+  case DrainNodeReason::DRAIN_NODE_REASON_IDLE_TERMINATION:
+    if (bool is_idle =
+            cluster_resource_scheduler_->GetLocalResourceManager().IsLocalNodeIdle();
+        is_idle) {
+      cluster_resource_scheduler_->GetLocalResourceManager().SetLocalNodeDraining(
+          request.reason());
       reply->set_is_accepted(true);
     } else {
       reply->set_is_accepted(false);
     }
-  } else {
+    break;
+  case DrainNodeReason::DRAIN_NODE_REASON_PREEMPTION:
     // Non-rejectable draining request.
-    RAY_CHECK_EQ(request.reason(),
-                 rpc::autoscaler::DrainNodeReason::DRAIN_NODE_REASON_PREEMPTION);
-    cluster_resource_scheduler_->GetLocalResourceManager().SetLocalNodeDraining();
+    cluster_resource_scheduler_->GetLocalResourceManager().SetLocalNodeDraining(
+        request.reason());
     reply->set_is_accepted(true);
+    break;
+  // Crash immediately.
+  case DrainNodeReason::DRAIN_NODE_REASON_UNSPECIFIED:
+  default:
+    RAY_LOG(ERROR) << "Drain raylet RPC has received, but the reason is not specified.";
```

Comment:
Why changing this? Before it will check fail but now it's only an error log

Reply:
Can we fix it?

Reply:
I fixed it